dlib/unicode/unicode.h

```diff
@@ -211,6 +211,112 @@ namespace dlib
             return -1;
         }
 
+        template <
+            typename charT,
+            typename forward_iterator
+            >
+        int u8_to_u32(
+            charT& result,
+            forward_iterator ibegin,
+            forward_iterator iend
+        )
```

Comment:
It would be good if we could get rid of the implementation above, but I saw it's used here: https://github.com/davisking/dlib/blob/257b81a17288edece7ffb8bbb5f00a7c2f473494/dlib/unicode/unicode.h#L259

Reply:
Should be able to make the above one call this new one by using `std::istream_iterator`.

Reply:
I had to use `std::istreambuf_iterator`, but it worked, thank you!